policies/recipes/artifacts.go

```diff
@@ -69,51 +67,52 @@ func FetchArtifacts(ctx context.Context, artifacts []*osconfigpb.SoftwareRecipe_
 
 func fetchArtifact(ctx context.Context, artifact *osconfigpb.SoftwareRecipe_Artifact, directory string) (string, error) {
 	localPath := filepath.Join(directory, artifact.Id)
-	uri, err := url.Parse(artifact.Uri)
-	if err != nil {
-		return "", fmt.Errorf("Could not parse url %q for artifact %q", artifact.Uri, artifact.Id)
-	}
-
 	var reader io.ReadCloser
-	switch strings.ToLower(uri.Scheme) {
-	case "gs":
-		reader, err = fetchWithGCS(ctx, uri.Host, uri.Path, uri.Fragment)
+	var err error
+	var checksum string
+	switch v := artifact.Artifact.(type) {
+	case *osconfigpb.SoftwareRecipe_Artifact_Gcs_:
+		reader, err = fetchWithGCS(ctx, v.Gcs.Bucket, v.Gcs.Object, v.Gcs.Generation)
 		if err != nil {
 			return "", fmt.Errorf("error fetching artifact %q from GCS: %v", artifact.Id, err)
 		}
 		defer reader.Close()
-	case "https", "http":
-		response, err := fetchWithHTTP(ctx, artifact.Uri)
+	case *osconfigpb.SoftwareRecipe_Artifact_Remote_:
+		if !strings.HasPrefix(v.Remote.Uri, "http") {
```

Comment:
Consider using the `url.Parse` code you have below to validate the scheme. It's safer and allows for a list `allowableRemoteArtifactSchems = [http, https]` which will be easier to maintain.